hatchways/team-gingerbread
# Justin Front-End/Integration Notifications

### What this PR does (required):
- fixes #54 
- if there are unread notifications, a green indicator appears next to the text in the "Notifications" button in LoggedInBar
- upon clicking the "Notifications" button, a dropdown is opened
- dropdown contains all unread notifications
- when the dropdown is opened, the readNotification helper is called to set each unread notification to read. also, the unread indicator is turned off
- if there are no unread notifications, the indicator is hidden, and opening the dropdown shows a "no new notifications" message
- updates to notifications BE: added middleware to get user's id via req.user.id instead of req.params.recipient, removed parameters from getAll and getUnread

### Screenshots / Videos (front-end only):
![image](https://user-images.githubusercontent.com/78042694/143726326-8220e0b5-f350-43eb-9ecc-25860972399c.png)
![image](https://user-images.githubusercontent.com/78042694/143726336-8ca2e5d9-0c58-4177-8582-af93adb000d7.png)
![image](https://user-images.githubusercontent.com/78042694/143726342-538d5951-3e85-4b4f-9066-8e580a6dcd2d.png)
![image](https://user-images.githubusercontent.com/78042694/143726348-2d786e0d-e553-47a6-9026-dd90f1cb1492.png)


### Any information needed to test this feature (required):
- create test notification in MongoDB according to the model specified in models/Notifications.js (set read to false)
- run server in terminal
```
npm run dev
```
- run client in terminal
```
npm start
```
- log in to the website
- click on "Notifications" button in the navbar, your test notification should appear in the dropdown
- close dropdown and go to MongoDB (your test notification read status should have changed from false to true)


### Any issues with the current functionality (optional):
- couldn't automatically merge, I believe I will have to rebase. I would appreciate some guidance on how to handle this error, as I am not an expert at Git/GitHub
- a filler notification is shown for a brief moment before the unread notifications load (had to include to prevent TypeScript errors)
